src/docfx/build/dependency/DependencyResolver.cs

```diff
@@ -145,7 +146,20 @@ public string GetRelativeUrl(Document fileRelativeTo, Document file)
             // Cannot resolve the file, leave href as is
             if (file is null)
             {
-                return (error, href, fragment, hrefType, null);
+                if (string.IsNullOrEmpty(pathToDocset))
+                {
+                    return (error, href, fragment, hrefType, null);
+                }
+
+                var (errorFromHistory, resourceFromHistory) = TryResolveResourceFromHistory(_gitCommitProvider, relativeTo.Docset, pathToDocset);
+                if (errorFromHistory != null || resourceFromHistory == null)
+                {
+                    return (errorFromHistory ?? error, href, fragment, hrefType, null);
```

Comment:
>errorFromHistory ?? error [](start = 28, length = 25)

if `errorFromHistory != null`, then the previous 'file-not-found' error will be overwritten?

Reply:
yes, then we want to throw the error why we can't get it from history, it a root cause why file is not found